modules/gapi/test/streaming/gapi_gstreamersource_tests.cpp

```diff
@@ -0,0 +1,401 @@
+
+
+TEST(GStreamerMultiSourceSmokeTest, Test)
+{
+    // Graph declaration:
+    cv::GMat in1, in2;
+    auto out = cv::gapi::add(in1, in2);
+    cv::GComputation c(cv::GIn(in1, in2), cv::GOut(out));
+
+    // Graph compilation for streaming mode:
+    auto ccomp = c.compileStreaming();
+
+    EXPECT_TRUE(ccomp);
+    EXPECT_FALSE(ccomp.running());
+
+    cv::gapi::wip::GStreamerPipeline
+        pipeline("videotestsrc is-live=true pattern=colors num-buffers=10 ! "
+                 "videorate ! videoscale ! "
+                 "video/x-raw,width=1920,height=1080,framerate=3/1 ! "
+                 "appsink name=sink1 "
+                 "videotestsrc is-live=true pattern=colors num-buffers=10 ! "
+                 "videorate ! videoscale ! "
+                 "video/x-raw,width=1920,height=1080,framerate=3/1 ! "
+                 "appsink name=sink2");
+
+    // GStreamer streaming sources configuration:
+    auto src1 = pipeline.getStreamingSource("sink1");
+    auto src2 = pipeline.getStreamingSource("sink2");
+
+    ccomp.setSource(cv::gin(src1, src2));
+
+    // Start of streaming:
+    ccomp.start();
+    EXPECT_TRUE(ccomp.running());
+
+    // Streaming - pulling of frames until the end:
+    cv::Mat in_mat_gapi;
+
+    EXPECT_TRUE(ccomp.pull(cv::gout(in_mat_gapi)));
+    EXPECT_TRUE(!in_mat_gapi.empty());
+    EXPECT_EQ(CV_8UC3, in_mat_gapi.type());
+
+    while (ccomp.pull(cv::gout(in_mat_gapi))) {
+        EXPECT_TRUE(!in_mat_gapi.empty());
+        EXPECT_EQ(CV_8UC3, in_mat_gapi.type());
+    }
+
+    EXPECT_FALSE(ccomp.running());
+    ccomp.stop();
+
+    EXPECT_FALSE(ccomp.running());
+}
+
+struct GStreamerMultiSourceTest :
+    public TestWithParam<std::tuple<cv::GComputation, cv::gapi::wip::GStreamerSource::OutputType>>
+{ };
+
+TEST_P(GStreamerMultiSourceTest, ImageDataTest)
+{
+    std::string pathToLeftIm = findDataFile("cv/stereomatching/datasets/tsukuba/im6.png");
+    std::string pathToRightIm = findDataFile("cv/stereomatching/datasets/tsukuba/im2.png");
+
+    std::string pipelineToReadImage("filesrc location=LOC ! pngdec ! videoconvert ! "
+        "videoscale ! video/x-raw,format=NV12 ! appsink");
+
+    cv::gapi::wip::GStreamerSource leftImageProvider(
+        std::regex_replace(pipelineToReadImage, std::regex("LOC"), pathToLeftIm));
+    cv::gapi::wip::GStreamerSource rightImageProvider(
+        std::regex_replace(pipelineToReadImage, std::regex("LOC"), pathToRightIm));
+
+    cv::gapi::wip::Data leftImData, rightImData;
+    leftImageProvider.pull(leftImData);
+    rightImageProvider.pull(rightImData);
+
+    cv::Mat leftRefMat =  cv::util::get<cv::Mat>(leftImData);
+    cv::Mat rightRefMat = cv::util::get<cv::Mat>(rightImData);
+
+    // Retrieve test parameters:
+    std::tuple<cv::GComputation, cv::gapi::wip::GStreamerSource::OutputType> params = GetParam();
+    cv::GComputation extractImage = std::move(std::get<0>(params));
+    cv::gapi::wip::GStreamerSource::OutputType outputType = std::get<1>(params);
```

Comment:
can `std::tie()` be used here instead?

```suggestion
    cv::GComputation extractImage;
    cv::gapi::wip::GStreamerSource::OutputType outputType = /*default value*/ 0;
    std::tie(extractImage, outputType ) = GetParam();
```

Is it because GComputation can't be declared like that?